### ISSUE TITLE
new DuringSplashScreen().runDuringSplashScreen() fails on first run

### ISSUE THREAD

**Issue by InI4** [osmdroid/osmdroid#1103]
## Issue Type

[x] Improvement

## Description and/or steps/code to reproduce the problem

When starting the `new DuringSplashScreen().runDuringSplashScreen()` on Splash of a freshly installed app, it seems to break, since the database is not yet existing. 

```
07-19 19:31:35.037 8155-8256/de.spieleck.app.badgers.debug E/SQLiteLog: (14) cannot open file at line 31278 of [2ef4f3a5b1]
    (14) os_unix.c:31278: (13) open(/storage/emulated/0/osmdroid/tiles/cache.db) -

```
Would be most convenient, if it checked for existence of the database first.

Currently it seems to work now when used like 
```
  onCreate(...) {
        ...
        AsyncSomething.execute(new Runnable() {
                @Override
                public void run() {
                    try {
                        new DuringSplashScreen().runDuringSplashScreen();
                    } catch ( Exception ignore ) { }
                }
        });
       ...
  }
```

simply 
```
         new DuringSplashScreen().runDuringSplashScreen();
```
would be nicer maybe.

### If it's a bug, version(s) of android this affects:

Probably All

### Version of osmdroid the issue relates to:

6.0.2

**Comment by monsieurtanuki** [comment]
@InI4 Couldn't reproduce this bug with the demo app, even when it was removed then reinstalled.
And when I have a look at the code, I don't understand in which cases it should happen when calling `new DuringSplashScreen().runDuringSplashScreen`:
````java
final SqlTileWriter sqlTileWriter = new SqlTileWriter();
sqlTileWriter.runDuringSplashScreen();
````
... as the `SqlTileWriter` constructor calls its `getDb` method, that precisely creates the `cache.db` file and the database.

I'm puzzled. Are you able to reproduce this bug on the demo app?

**Comment by spyhunter99** [comment]
maybe remove it, then ensure the cache.db is deleted and retry?

**Comment by monsieurtanuki** [comment]
@spyhunter99 You're right, I assumed that the `cache.db` file was removed by the app uninstall and it's not the case.
I tried again, removed the demo app and the `cache.db` file, but still couldn't find any crash.
This is the piece of code I edited for the test, in class `IntroActivity`:
````java
@Override
public void onCreate(Bundle savedInstanced){
    super.onCreate(savedInstanced);
    new DuringSplashScreen().runDuringSplashScreen(); // added code
...
````

@InI4 Could it be that you are talking about a new app _that is not osmdroid demo_, and that this new app hasn't got the privileges to create the `cache.db` file?

Still puzzled...

**Comment by monsieurtanuki** [comment]
@InI4 I just wanted to be able to reproduce this issue, and the demo app was appropriate for that.
I don't know that much about device/android version issues.
The best you can do is perhaps to track step by step what goes wrong, given that:
* from reading the code, I can say it's not supposed to happen
* I was not able to reproduce this issue
